### PR TITLE
Add new sample functionality

### PR DESCRIPTION
**Pull request by gabrysgab**
### Description
Change sample activity to show atm's in the area
Architecture changes
Added necessary dependencies

### CODE REVIEW

**Comment by netguru-hound** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/ar-localizer-view-android/blob/12a3975346c52033898d1c094f2ce29cf815f4ca/app/src/main/kotlin/com/netguru/android/arlocalizeralternative/feature/location/presentation/LocationActivity.kt#L34">app/src/main/kotlin/com/netguru/android/arlocalizeralternative/feature/location/presentation/LocationActivity.kt#L34</a> - Class ‘LocationActivity’ with ‘21’ functions detected. Defined threshold inside classes is set to ‘15’</td>
    </tr>
  </tbody>
</table>

### JaCoCO Code Coverage 21.57% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|


#### APK file size

| size |
| --- |
| 3.3MB |



#### Number of method references

| Dex file | count |
| --- | --- |
| classes.dex | 29017 |



<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
     >
 
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+    <uses-permission android:name="android.permission.INTERNET"/>
 
     <application
```
Comment (netguru-hound) [inline app/src/main/AndroidManifest.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/AndroidManifest.xml#L9"></span>On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute <code>android:fullBackupContent</code> to specify an <code>@xml</code> resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html</td>
    </tr>
  </tbody>
</table>

Comment (netguru-hound) [inline app/src/main/AndroidManifest.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/AndroidManifest.xml#L9"></span>App is not indexable by Google Search; consider adding at least one Activity with an ACTION-VIEW intent filter. See issue explanation for more details.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/values/colors.xml [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
     <color name="colorPrimary">#008577</color>
```
Comment (netguru-hound) [inline app/src/main/res/values/colors.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/res/values/colors.xml#L6"></span>The resource <code>R.color.colorPrimary</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/values/colors.xml [hunk 3]
```diff
@@ -5,6 +5,7 @@
...
     <color name="colorPrimaryDark">#00574B</color>
```
Comment (netguru-hound) [inline app/src/main/res/values/colors.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/res/values/colors.xml#L7"></span>The resource <code>R.color.colorPrimaryDark</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -5,8 +5,7 @@
     <string name="destination_location_label">Destination Location</string>
```
Comment (netguru-hound) [inline app/src/main/res/values/strings.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/res/values/strings.xml#L5"></span>The resource <code>R.string.destination_location_label</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -5,8 +5,7 @@
...
     <string name="longitude_label">Longitude</string>
```
Comment (netguru-hound) [inline app/src/main/res/values/strings.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/res/values/strings.xml#L6"></span>The resource <code>R.string.longitude_label</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -5,8 +5,7 @@
...
     <string name="latitude_label">Latitude</string>
```
Comment (netguru-hound) [inline app/src/main/res/values/strings.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/res/values/strings.xml#L7"></span>The resource <code>R.string.latitude_label</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -5,8 +5,7 @@
...
-    <string name="netguru_offices_label">Netguru Offices</string>
-    <string name="gdansk_office_label">Gdansk Office</string>
+    <string name="find_atm_label">Find ATM</string>
     <string name="latitude_error">Latitude value should be between -90 an 90</string>
```
Comment (netguru-hound) [inline app/src/main/res/values/strings.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/res/values/strings.xml#L9"></span>The resource <code>R.string.latitude_error</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -5,8 +5,7 @@
...
     <string name="longitude_error">Longitude value should be between -180 an 180</string>
```
Comment (netguru-hound) [inline app/src/main/res/values/strings.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/res/values/strings.xml#L10"></span>The resource <code>R.string.longitude_error</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -5,8 +5,7 @@
...
     <string name="empty_value_error">Value should not be empty</string>
```
Comment (netguru-hound) [inline app/src/main/res/values/strings.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/6e7a43f049fda9487b1419ac862329af39e581fa/app/src/main/res/values/strings.xml#L11"></span>The resource <code>R.string.empty_value_error</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/com/netguru/android/arlocalizeralternative/feature/location/domain/ILocationRepository.kt [hunk 10]
```diff
@@ -0,0 +1,16 @@
+package com.netguru.android.arlocalizeralternative.feature.location.domain
+
+import com.google.android.gms.maps.model.LatLng
+import com.netguru.arlocalizerview.location.LocationData
+import io.reactivex.Single
+
+interface ILocationRepository {
+    fun getDestinations(
+        amenity: String, southWestLatitude: Double,
```
Comment (GabrielSamojlo) [inline app/src/main/kotlin/com/netguru/android/arlocalizeralternative/feature/location/domain/ILocationRepository.kt]: move `southWestLatitude: Double` to newline to be consistent with other parameters

File: app/src/main/kotlin/com/netguru/android/arlocalizeralternative/feature/location/presentation/LocationViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.rxkotlin.subscribeBy
+import io.reactivex.schedulers.Schedulers
+import javax.inject.Inject
+
+class LocationViewModel @Inject constructor(
+    private val atmUseCase: AtmUseCase,
+    currentLocationUseCase: CurrentLocationUseCase) :
```
Comment (GabrielSamojlo) [inline app/src/main/kotlin/com/netguru/android/arlocalizeralternative/feature/location/presentation/LocationViewModel.kt]: move parenthesis to the new line - we should stick to official code convention

File: app/src/test/kotlin/com/netguru/android/arlocalizeralternative/feature/location/LocationViewModelTest.kt [hunk 12]
```diff
@@ -34,17 +47,21 @@ class LocationViewModelTest {
 
     @Before
     fun init() {
-        viewModel = LocationViewModel(fusedLocationProviderClient)
-        whenever(fusedLocationProviderClient.lastLocation).thenReturn(locationTask)
+        whenever(currentLocationUseCase.getCurrentLocation()).thenReturn(Single.just(currentLocation))
+        viewModel =
+            LocationViewModel(
+                atmUseCase, currentLocationUseCase
+            )
```
Comment (GabrielSamojlo) [inline app/src/test/kotlin/com/netguru/android/arlocalizeralternative/feature/location/LocationViewModelTest.kt]: why not keep it in a single line?

File: app/src/main/kotlin/com/netguru/android/arlocalizeralternative/feature/location/data/LocationRepository.kt [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+                    }
+                } else {
+                    emitter.onError(Throwable(LOCATION_ERROR))
+                }
+            }
+        }
+    }
+
+    override fun getDestinations(
+        amenity: String, southWestLatitude: Double,
```
Comment (GabrielSamojlo) [inline app/src/main/kotlin/com/netguru/android/arlocalizeralternative/feature/location/data/LocationRepository.kt]: move `southWestLatitude: Double` to newline

**Review by GabrielSamojlo — APPROVED** [review]

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (netguru-hound) [inline app/src/main/AndroidManifest.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/8923affeef1599c8bc3e0684d14989e5173c91ea/app/src/main/AndroidManifest.xml#L9"></span>On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute <code>android:fullBackupContent</code> to specify an <code>@xml</code> resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html</td>
    </tr>
  </tbody>
</table>

Comment (netguru-hound) [inline app/src/main/AndroidManifest.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/ar-localizer-view-android/blob/8923affeef1599c8bc3e0684d14989e5173c91ea/app/src/main/AndroidManifest.xml#L9"></span>App is not indexable by Google Search; consider adding at least one Activity with an ACTION-VIEW intent filter. See issue explanation for more details.</td>
    </tr>
  </tbody>
</table>

**Pull request merged**